src/zenml/zen_server/routers/users_endpoints.py

```diff
@@ -235,11 +250,30 @@ def update_user(
 
         Returns:
             The updated user.
+
+        Raises:
+            IllegalOperationError: if the user tries change admin status,
+                while not an admin
         """
         user = zen_store().get_user(user_name_or_id)
         if user.id != auth_context.user.id:
-            verify_permission_for_model(user, action=Action.UPDATE)
+            verify_admin_status_if_no_rbac(
+                auth_context.user.is_admin, "update other user"
+            )
+            verify_permission_for_model(
+                user,
+                action=Action.UPDATE,
+            )
+        if (
+            user.is_admin != user_update.is_admin
+            and not auth_context.user.is_admin
+        ):
+            raise IllegalOperationError(
+                "Only admins can change the admin status of other users."
+            )
 
+        user_update.activation_token = user.activation_token
+        user_update.active = user.active
```

Comment:
```suggestion
        if not auth_context.user.is_admin or user.id == auth_context.user.id:
            user_update.active = user.active
```

An admin is allowed to activate or deactivate another account without using the `activate` and `deactivate` endpoints (i.e. without involving activation tokens).

Reply:
Sorry, I'm lost: how this suggestion is helping? You advised to check for the user in auth **is not** an admin or is the same user and only if this is achieved set active update. Why isn't the user in auth **is** admin or is the same user then?

Reply:
I set it to `if auth_context.user.is_admin or user.id == auth_context.user.id:`, hope this is what's expected

Reply:
Let me explain the change I suggested: if a user is not admin, or if the user is trying to modify their own account, ignore the `active` setting in the request.

Reply:
Yeah, now I got it 🙂 Thanks!
I made a change